jax/numpy/lax_numpy.py

```diff
@@ -3788,6 +3789,39 @@ def _quantile(a, q, axis, interpolation, keepdims):
   return lax.convert_element_type(result, a.dtype)
 
 
+@partial(jit, static_argnums=2)
+@partial(vmap, in_axes=(None, 0, None))
+def _searchsorted(a, v, side):
+  op = operator.le if side == 'left' else operator.lt
+
+  def cond_fun(state):
+    start, stop = state
+    return stop - start > 1
+
+  def body_fun(state):
+    start, stop = state
+    mid = (start + stop) // 2
+    return where(op(v, a[mid]), (start, mid), (mid, stop))
+
+  result = lax.while_loop(cond_fun, body_fun, array([0, a.shape[0]]))
```

Comment:
Have you considered implementing the loop over two separate `start` and `stop` arrays instead of a combined array?

I would be interested to see if it makes a performance difference. In particular, I suspect it might be considerably more efficient on TPUs.

Reply:
I'm not certain what you mean by two separate start and stop arrays

Reply:
Something like:
```python
  def body_fun(state):
    start, stop = state
    mid = (start + stop) // 2
    go_left = op(v, a[mid])
    return where(go_left, start, mid), where(go_left, mid, stop)

  result = lax.while_loop(cond_fun, body_fun, (0, a.shape[0])))
```

The different here is that `state` is a tuple of (start, stop) of an array with shape `(2,)`

Reply:
Done - though this doesn't seem to make a noticeable difference in computation time, I think it's a bit cleaner.